index.js

```diff
@@ -87,10 +74,46 @@ module.exports = {
     return browserTree;
   },
 
+  //If user don't set  disableDefaultNodeFetch to be true,
   //add node version of fetch.js into fastboot package.json manifest vendorFiles array
-  updateFastBootManifest: function (manifest) {
-    manifest.vendorFiles.push('ember-fetch/fastboot-fetch.js');
+  updateFastBootManifest: function(manifest) {
+    if (this._shouldImportDefaultNodeModule()) {
+        manifest.vendorFiles.push('ember-fetch/fastboot-fetch.js');
+    }
     return manifest;
+  },
+  
+  //return host app object
+  _findHostApp: function() {
+    // If the addon has the _findHost() method (in ember-cli >= 2.7.0), we'll just
+    // use that.
+    let target;
+    if (typeof this._findHost === 'function') {
+        target = this._findHost();
+    }
+
+    // Otherwise, we'll use this implementation borrowed from the _findHost()
+    // method in ember-cli.
+    // Keep iterating upward until we don't have a grandparent.
+    // Has to do this grandparent check because at some point we hit the project.
+    let current = this;
+    do {
+        target = current.app || app;
+    } while (current.parent.parent && (current = current.parent));
+    return target;
+  },
+
+  //convenient method to determine if we should import the default node fetch module
+  //or skip. Let consumer app to import node fetch from elsewhere
+  _shouldImportDefaultNodeModule: function() {
+    var app = this._findHostApp();
+    var appOptions = (app && app.options) || {};
+    //make it backwards compatiable for apps doesn't set this config
+    if (typeof appOptions['ember-fetch'] === 'undefined') {
```

Comment:
I'd suggest changing this conditional around a bit.  As it stands, setting `'ember-fetch'` to `{}` in `ember-cli-build.js` will disable the default behavior, but I don't think that is what we want.  I believe we would want to only disable if the value is explicitly `true`.

Reply:
This utility function determines if we should define fetch module from node-fetch. It's reverse from the configuration flag disableDefaultNodeFetch. If _shouldImportDefaultNodeModule return true, node-fetch is define. If return false, it's not defined.

The logic is a bit flipped here because I want to make it backwards compatible. How should I make it more clear?

Reply:
What I'm saying is that the logic is not quite right, you get node-fetch disabled in both of the following cases:

1. you set `'ember-fetch': { disableDefaultNodeFetch: true }`
2. you have `'ember-fetch': {}`